@xen-orchestra/web-core/lib/components/network/VtsNetworkStatus.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+  <UiInfo class="status-indicator text-ellipsis" :accent="computedStatusProps.accent">
+    <p v-tooltip class="text-ellipsis" :class="{ card }">{{ computedStatusProps.text }}</p>
+  </UiInfo>
+</template>
+
+<script setup lang="ts">
+import UiInfo from '@core/components/ui/info/UiInfo.vue'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { faCheck, faExclamation } from '@fortawesome/free-solid-svg-icons'
+import { computed } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const props = defineProps<{
+  items?: any | any[]
+  card?: boolean
+}>()
+
+const { t } = useI18n()
+
+type Status = 'connected' | 'disconnected' | 'partially_connected'
+type StatusAccent = 'success' | 'warning' | 'danger'
+
+const states: Record<Status, { text: string; icon?: IconDefinition; accent: StatusAccent }> = {
+  connected: { text: t('connected'), icon: faCheck, accent: 'success' },
+  disconnected: { text: t('disconnected'), icon: faCheck, accent: 'danger' },
+  partially_connected: { text: t('partially-connected'), icon: faExclamation, accent: 'warning' },
+}
+
+const items = computed(() => (Array.isArray(props.items) ? props.items : props.items ? [props.items] : []))
```

Comment:
perhaps we can make this easier to read
```suggestion
const items = computed(() => {
  if (!props.items) return []
  return Array.isArray(props.items) ? props.items : [props.items]
})
```